code/WorkInProgress/DrMelonsStuff.dm

```diff
@@ -114,117 +147,167 @@
 		. += "<br><span class='notice'>[reagents.get_description(user,RC_FULLNESS|RC_VISIBLE|RC_SPECTRO)]</span>"
 		return
 
-	on_reagent_change()
-		..()
-		src.overlays = null
+	// using overlay levels so it looks like you're in the bath
+	// we don't use show_submerged_image since we want the head to poke out
+	// MOB_LAYER(-0) =  all other mobs on top of us
+	// MOB_LAYER-0.1 = Bath edge to trim feet
+	// MOB_LAYER-0.2 = Water overlay
+	// MOB_LAYER-0.3 = Occupant
+	// MOB_LAYER-0.4 = Water underlay
 
+	on_reagent_change()
 		if(reagents.total_volume)
-			ENSURE_IMAGE(src.fluid_icon, src.icon_state, "fluid-bathtub")
+			var/image/new_underlay = src.SafeGetOverlayImage("fluid_underlay", 'icons/obj/stationobjs.dmi', "fluid_bathtub", MOB_LAYER - 0.4)
 			var/datum/color/average = reagents.get_average_color()
-			fluid_icon.color = average.to_rgba()
-			src.overlays += fluid_icon
-
-		if(src.myuser)
-			src.overlays += src.myuser
-			// Don't really see the purpose of this; re-add if you believe otherwise
-			// if(reagents.total_volume)
-				// src.overlays += fluid_icon_feet
-
-		src.overlays += bath_edge
+			var/percent_filled = reagents.total_volume / reagents.maximum_volume
```

Comment:
this isn't really a percent (should be filled_ratio or fraction_filled or something)